source/github/README.md

```diff
@@ -1,11 +1,12 @@
 # github
 
-`github://user:personal-access-token@owner/repo/path#ref`
+`github://user:personal-access-token@github.com/owner/repo/path#ref`

```

Comment:
This should be: `github://user:personal-access-token@hostowner/repo/path#ref`

Reply:
@dhui I'm a bit unsure on how to differentiate:

```
github://user:personal-access-token@owner/repo/path#ref (use public github)

vs

github://user:personal-access-token@enterprise.myhost.com/owner/repo/path#ref (use enterprise github)
```

Any ideas?

Reply:
IDK of a clean way to differentiate.

Options I see:
1. We could try to detect if the host part looks like a domain and if it doesn't, default to github.com but that's quite dirty... e.g. `localhost` won't look like a domain

1. Another *slightly* more robust option would be to try both the host **and** github.com. My main concern w/ this is leaking credentials. We'd also have to implement the fallback logic and log errors.

1. Add a new github source driver where the host must be specified.
We could then replace the old github source driver with the new one when we're ready to make the backwards incompatible change.

1. Make a backwards incompatible change. e.g. require the host to be specified for the current github source driver and break existing source paths.

Do you have any ideas for other options or thoughts on the ones listed above?

Reply:
@dhui How about I make the driver for Github Enterprise, that reuses github one, but would obviously resolve the GH client differently. So essentially, the `github` one wouldn't change and the usage would remain the same. That feels like the _cleanest_ approach right now.

Reply:
I agree that using a new driver is the cleanest approach for now.

What scheme are you thinking about using to register the driver? Maybe `github2`? I'd like a name that's more descriptive but also short. `githube` or `github-ent` might work. Is the driver limited to enterprise or will any domain work?

Reply:
I was thinking something like `github-ee` (Enterprise Edition). The full URL scheme would be `github-ee://username:token@myhost.com/owner/repo/path#ref` `myhost.com` can be any host where Github Enterprise instance is hosted. Github Go Client requires an API host and Uploads, which I believe by default sit under `api.` and `uploads.` subdomains.

Reply:
`github-ee` sounds good to me. Make sure you document the differences in both source drivers' README files.

We may need to provide a way to customize the api and upload subdomains, but I think it's fine to do that later.

Reply:
I'll crack on then